Tests/AblyChatTests/Mocks/MockRealtimeChannel.swift

```diff
@@ -29,7 +29,7 @@ final class MockRealtimeChannel: InternalRealtimeChannelProtocol {
         initialErrorReason: ARTErrorInfo? = nil,
         attachBehavior: AttachOrDetachBehavior? = nil,
         detachBehavior: AttachOrDetachBehavior? = nil,
-        messageJSONToEmitOnSubscribe: [String: Sendable]? = nil,
+        messageJSONToEmitOnSubscribe: [String: JSONValue]? = nil,
```

Comment:
As mentioned in #249 I don't understand the purpose of this `messageJSONToEmitOnSubscribe`. But also it seems like it would make sense to just use `JSONValue` in the first place in #249?

Reply:
removed, using `simulateIncomingMessage` instead